src/rgw/rgw_admin.cc

```diff
@@ -8108,7 +8108,34 @@ int main(int argc, const char **argv)
       cerr << "ERROR: sync.init() returned ret=" << ret << std::endl;
       return -ret;
     }
-    ret = sync.read_sync_status(dpp());
+
+    rgw_bucket_sync_status full_status;
+    RGWBucketSyncPolicyHandlerRef handler;
+    const auto& sources = handler->get_sources();
+    const auto source_it = sources.find(source_zone);
+    if (source_it == sources.end()) {
+      cerr << "ERROR: source zone: '" << source_zone << "' could not be found" << std::endl;
+      return 0;
+    }
+    const auto& pipes = source_it->second;
+    if (pipes.pipe_map.empty()) {
+      cerr << "ERROR: no destination zones for source zone: '" << source_zone << "'" << std::endl;
+      return 0;
+    }
+    const auto& pipe = pipes.pipe_map.begin()->second;
+
+    ret = rgw_read_bucket_full_sync_status(dpp(), static_cast<rgw::sal::RadosStore*>(store), pipe, &full_status, null_yield);
+    if (ret < 0) {
+      cerr << "ERROR: reading bucket full sync status returned ret= " << ret << std::endl;
+      return -ret;
+    }
+
+    if (full_status.state != BucketSyncState::Incremental) {
+      cerr << "cannot sync markers for non incremental status= " << full_status.state << std::endl;
+      return 0;
+    }
```

Comment:
the `bucket sync markers` command just prints the current sync status, so i think it would make sense to dump the json format of both the full- and incremental status, similar to how Adam's `bilog_status_v2::dump()` does for `RGWOp_BILog_Status`

so even if `state != Incremental`, we can still dump the "full_status" along with an empty map for "sync_status"

Reply:
there is already a `BUCKET_SYNC_STATUS` command. do you want `BUCKET_SYNC_MARKERS` to do the same thing?

Reply:
not needed anymore